### PR TITLE
grades

### PR DESCRIPTION
**Pull request by gjcritchlow21**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,18 +0,0 @@
...
-
-start();
-
-async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
-  process.exit();
-}
```
Comment (gjcritchlow21) [inline index.js:18]: In the future for organization of your projects, delete any files that you are not gonna to use. It cleans up your work and lessens the chance that you publish any unwanted or secure code.

File: projectnumber.js [hunk 2]
```diff
@@ -1,84 +0,0 @@
...
-
-async function start() {
-    let maxValue
-    let rangeQuestion= await ask("Would you like to extend the number range?") 
-    
-    if (rangeQuestion==='yes'||rangeQuestion==='yea') {
-        rangeQuestion=await ask ('What range would you like to set?')
-    }
-   
-    var extendedRange=parseInt(rangeQuestion)
```
Comment (gjcritchlow21) [inline projectnumber.js:18]: I would recommend against using `var` and instead use `let` and also be consistent in how you use them. `var` has a problem of allowing scope creep while `let` can only be accessed in the scope its declared.

File: projectnumber.js [hunk 3]
```diff
@@ -1,84 +0,0 @@
-const readline = require('readline');
```
Comment (gjcritchlow21) [inline projectnumber.js:1]: Make sure you format your code before your final push up

File: projectnumber.js [hunk 4]
```diff
@@ -1,84 +0,0 @@
...
-        readlineInterface.question(questionText, resolve);
-    });
-}
-
-async function start() {
-    let maxValue
-    let rangeQuestion= await ask("Would you like to extend the number range?") 
-    
-    if (rangeQuestion==='yes'||rangeQuestion==='yea') {
-        rangeQuestion=await ask ('What range would you like to set?')
```
Comment (gjcritchlow21) [inline projectnumber.js:15]: Your questioning is kind of clunky here, am I setting the min value also?

File: projectnumber.js [hunk 5]
```diff
@@ -1,84 +0,0 @@
...
-        
-        guessArray.push(guess) 
-        if (guessArray===guess){
-            console.log('same guess')
-        }
-        
-        console.log('Hmmm...' + first + " " + "you say.")
-        
-        if (first === 'greater') {
-            minValue = guess
```
Comment (gjcritchlow21) [inline projectnumber.js:50]: When you reset the min and max, you want to make sure that the guess does not become either/or. For instance, if your number is higher then a guess of 25, you want the new range to be 26-100 - and if its lower, you would want 1-24.

File: projectnumber.js [hunk 6]
```diff
@@ -1,84 +0,0 @@
...
-        }
-        guessCounter++
-       
-        if (guessCounter>=9) {
-            console.log("You've reached the max number of guesses.")
-            process.exit()
-        }
-        
-        else if (first === 'equal to' || first ==='equal') {
-            response=await ask ("I thought I would win, better luck next time!")
```
Comment (gjcritchlow21) [inline projectnumber.js:66]: Your game ends and hangs right here. This is because your program is waiting for input for this response. If you get rid of the await ask, the `process.exit()` will get hit and end your game.

File: projectnumber.js [hunk 7]
```diff
@@ -1,84 +0,0 @@
...
-    while (question === 'yes' || question=== 'yea') {
-      //check if valid range min=max=cheat  
-        let guess= Math.floor((maxValue+minValue) / 2)
-       
-        {let first = await ask('Is your number greater, less than , or equal to' + ' ' + guess + '? ')
-        
-        guessArray.push(guess) 
-        if (guessArray===guess){
-            console.log('same guess')
-        }
```
Comment (gjcritchlow21) [inline projectnumber.js:45]: Your cheat detector only checks if something has been guessed twice, which wont be the case because the min and max values will be reset. The player should be prompted to input their guess so that you can check against that variable and determine if the player is cheating. Something along the lines of:
`let playerNumber = await ask('What is your number? I wont cheat")`
then you check that number against what the computer guesses, the range values and your guess array.

File: computerAsk.js [hunk 8]
```diff
@@ -1,85 +0,0 @@
-//computer stores number between1-100
```
Comment (gjcritchlow21) [inline computerAsk.js:1]: Don't forget to format your code before you push it up

File: computerAsk.js [hunk 9]
```diff
@@ -1,85 +0,0 @@
...
-    
-    
-    let rangeQuestion= await ask("Hi there, would you like to extend the number range?") 
-    
-    
-    function randomInt(minValue, maxValue) {
-        let extendedRange = maxValue - minValue + 1; 
-        return minValue + Math.floor(Math.random() * extendedRange);
-      }
-      var secretNumber=parseInt(randomInt)
```
Comment (gjcritchlow21) [inline computerAsk.js:25]: See comments about the use of `var` and `let` in your projectnumber file below.

File: computerAsk.js [hunk 10]
```diff
@@ -1,85 +0,0 @@
...
-            console.log("You've reached the max number of guesses.")
-            process.exit()
-             }
-            
-             else if (first === secretNumber) {
-            response=await ask ("You've won!")
-            process.exit()
-            }
-        }    
-        }
```
Comment (gjcritchlow21) [inline computerAsk.js:81]: The computer guesses only once, then hits line 67 and hangs there.  This is because the loop is checking the guess on lines 55 and 57 and printing to the console, but since there has been only one guess, it doesn't go into the if statement on line 68.  You should move the else if statement on 76 into the if loop in 53.

**Review by gjcritchlow21 — COMMENTED** [review]
Functionality: There are a few wrinkles that need to be smoothed out in projectnumber.js and the cheat detector needs work, however the game works and you should be proud of what you've done. The computerAsk.js is broken but is just a little bit of tinkering and rearranging away from functioning as it should. 

Organization:  Your code is organized well, but get rid of the clutter and reduce some of the spacing throughout. Also if you are not going to use it, delete it (ie index.js)

Readability: FORMAT YOUR CODE! and also add more notes,

File: computerAsk.js (hunk 8 above)
Comment (brendan802) [reply computerAsk.js:1]: roger that thanks GJ

File: computerAsk.js (hunk 10 above)
Comment (brendan802) [reply computerAsk.js:81]: thank you that makes complete sense!

File: projectnumber.js (hunk 4 above)
Comment (brendan802) [reply projectnumber.js:15]: Ok, I see that. I didn't plan to have the player also set the min, but that could be refined definitely

File: projectnumber.js (hunk 6 above)
Comment (brendan802) [reply projectnumber.js:66]: so just a process.exit()("string")?

**Review by brendan802 — COMMENTED** [review]
Thanks for all the constructive tips GJ quite thorough.
